contributed/Source+Catalog+Emulation+with+CosmoDC2Realizer.ipynb

```diff
@@ -0,0 +1,535 @@
+    "        all_gal, all_gal_idx = helper.get_neighbors(galaxies_in_field, ra_center, dec_center, deblending_scale) \n",
+    "        num_gal_neighbors = len(all_gal_idx) - 1 # subtract central galaxy itself\n",
+    "        # Stars/AGN neighbors (point)\n",
+    "        point, point_idx = helper.get_neighbors(points_in_field, ra_center, dec_center, deblending_scale) \n",
+    "        num_star_neighbors = point['star'].sum()\n",
+    "        num_agn_neighbors = point['agn'].sum()\n",
+    "        num_sprinkled_neighbors = point['sprinkled'].sum()\n",
+    "\n",
+    "        #################\n",
+    "        # Sersic to MoG #\n",
+    "        #################\n",
+    "        # Separate galaxy catalog into bulge and disk\n",
+    "        bulge, disk, all_gal = helper.separate_bulge_disk(all_gal)\n",
+    "        # Deconstruct bulge/disk into MoG\n",
+    "        bulge_mog = helper.sersic_to_mog(sersic_df=bulge, bulge_or_disk='bulge')\n",
+    "        disk_mog = helper.sersic_to_mog(sersic_df=disk, bulge_or_disk='disk')\n",
+    "        point_mog = helper.point_to_mog(point_df=point)\n",
+    "        # Concat the three\n",
+    "        full_mog = pd.concat([bulge_mog, disk_mog, point_mog], axis=0)\n",
+    "        \n",
+    "        # Add some metadata\n",
+    "        full_mog['objectId'] = central_gal['galaxy_id'] # identifier for blended system\n",
+    "        full_mog['num_gal_neighbors'] = num_gal_neighbors\n",
+    "        full_mog['num_star_neighbors'] = num_star_neighbors\n",
+    "        full_mog['num_agn_neighbors'] = num_agn_neighbors\n",
+    "        full_mog['num_sprinkled_neighbors'] = num_sprinkled_neighbors\n",
+    "        \n",
+    "        # Join with observations\n",
+    "        mog_observed = helper.join_with_observation(full_mog, obs_history_in_field)\n",
+    "        # Remove unobserved fluxes (only keep flux of observed filter)\n",
+    "        mog_observed = helper.collapse_unobserved_fluxes(mog_observed)\n",
+    "        # Calculate moment contribution of each Gaussian of blended system\n",
+    "        mog_observed = moments.calculate_total_flux(mog_observed)\n",
+    "        mog_observed = moments.calculate_1st_moments(mog_observed)\n",
+    "        mog_observed = moments.calculate_2nd_moments(mog_observed)\n",
+    "        # Collapse MoGs into one blended system\n",
+    "        blended = moments.collapse_mog(mog_observed)\n",
+    "        blended = moments.apply_environment(blended)\n",
+    "        \n",
+    "        source_in_field = pd.concat([source_in_field, blended], axis=0)\n",
+    "        if (gal_idx + 1) %100 == 0:\n",
+    "            print(\"%d systems completed\" %(gal_idx + 1))\n",
+    "        \n",
+    "    source = pd.concat([source, source_in_field], axis=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here are sample light curves of a blended system consisting of 2 galaxies."
+   ]
```

Comment:
Here it would be great to have some more info for what to expect (particular for those who are new to this). To show the impact of the blending, could you show the light curves of the galaxies separately too on a different plot?

Reply:
Done!